mlflow/pipelines/utils/metrics.py

```diff
@@ -41,13 +41,10 @@ def from_custom_metric_dict(cls, custom_metric_dict):
     PipelineMetric(name="false_positives", greater_is_better=False),
     PipelineMetric(name="false_negatives", greater_is_better=False),
     PipelineMetric(name="true_positives", greater_is_better=True),
-    PipelineMetric(name="recall", greater_is_better=True),
-    PipelineMetric(name="precision", greater_is_better=True),
+    PipelineMetric(name="recall_score", greater_is_better=True),
+    PipelineMetric(name="precision_score", greater_is_better=True),
     PipelineMetric(name="f1_score", greater_is_better=True),
     PipelineMetric(name="accuracy_score", greater_is_better=True),
-    PipelineMetric(name="log_loss", greater_is_better=False),
-    PipelineMetric(name="roc_auc", greater_is_better=True),
-    PipelineMetric(name="precision_recall_auc", greater_is_better=True),
```

Comment:
why do we need to remove this? it doesn't look like we are adding these back either. i'm not following.

Reply:
That's a great question, and one that we should probably follow up on.

When I attempted to use `log_loss` as my primary metric, the train step failed and returned an error that included the list of allowed metric names. The list did not include the three metrics removed here.

Reply:
Were there other builtin metric names included in the list that aren't captured here? Where was the list of builtins printed from?

Reply:
This is the error thrown by the evaluate step if `log_loss` is included in this list and specified in `pipeline.yaml`:
```
mlflow.exceptions.MlflowException: The metric log_loss is defined in the pipeline's validation criteria but was not returned from mlflow evaluation.
```

Here's a warning from the train step:
```
2022/10/21 12:25:41 WARNING mlflow.pipelines.utils.step: Input metric names {'log_loss'} not found in eval metrics: {'score', 'precision_score', 'false_negatives', 'true_positives', 'accuracy_score', 'recall_score', 'f1_score', 'example_count', 'true_negatives', 'false_positives'}
```

`example_count` and `score` are not captured here, let me know if you'd like to see them added. And also if you'd like me to create a bug report here (unless I've missed something obvious, quite possible!)